### PR TITLE
Enhancement/v3 event service

### PR DESCRIPTION
**Pull request by Wardormeur**
## Description

set a "manual" proxy with validation for event service
set a default reusable transport
set a default reusable query validation

## Motivation and Context

https://trello.com/c/x59K257I

## How Has This Been Tested?

Unit tests + manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I gave a thought about security (XSS, perms, injection of params, scoping, ...)
- [x] Linting passes

## Reviewer checklist:

 - [ ] Security
   - [x] Permissions
   - [x] Data scope and leaks
   - [x] Injection
   - [x] XSS
 - [ ] Performance
   - [x] tested with production data
 - [ ] Code
   - [x] Single responsibility
   - [x] Naming
 - [ ] Design
   - [ ] Responsive
   - [ ] Scale well with production data
   - [ ] Looks great !

### CODE REVIEW

File: web/lib/handlers/event.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const mastermind = require('../mastermind');
+const Event = require('../models/event');
+
+const get = params => // eslint-disable-line no-unused-vars
+  mastermind([
+    // eslint-disable-next-line no-unused-vars
+    async (req, reply, cb) => {
```
Comment (butlerx) [inline web/lib/handlers/event.js]: using async without a try catch is deprecated in node 

https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections

Comment (Wardormeur) [reply web/lib/handlers/event.js]: it's caught by mastermind, which is an async series

Comment (butlerx) [reply web/lib/handlers/event.js]: This https://github.com/CoderDojo/cp-zen-platform/blob/master/web/lib/mastermind.js?

I can't see how that would handle a rejected promise?

Comment (Wardormeur) [reply web/lib/handlers/event.js]: See https://caolan.github.io/async/index.html "Using ES2017 async functions"

Comment (butlerx) [reply web/lib/handlers/event.js]: thats was added in async 2.3 though this project still uses 0.9

Comment (Wardormeur) [reply web/lib/handlers/event.js]: Good catch

Comment (butlerx) [reply web/lib/handlers/event.js]: https://github.com/caolan/async/blob/v1.5.2/README.md#asyncify could work if you dont want to upgrade async  all the way to v2

**Pull request merged**